### ISSUE TITLE
Remove akka

### ISSUE THREAD

**Issue by fommil** [ensime/ensime-server#1634]
The continuation of #1351

**Comment by fommil** [comment]
I keep coming back to core.async http://blog.podsnap.com/scasync.html

**Comment by zainab-ali** [comment]
So here's my two pence:
If we want to transition to a stream based solution, we shouldn't so much be asking "what library should we use?", but "what will my streams look like?".   Something like [swave's graphs](http://swave.io/usage/show-off/fibonacci.html) showing where we want to debounce, where the async boundaries are etc.  I've taken a quick glance through the code, but creating something like this would take me some more time.

Once we have such a diagram, we can decide whether it's best to do in fs2 / monix / swave, all of which are slightly different.

That's if we do want a stream based solution.  Core async looks interesting, but I can't find a popular Scala implementation.

**Comment by fommil** [comment]
Did you read the blog post and impl linked above? It's definitely the starting point for core.async. go for it!

**Comment by yilinwei** [comment]
@fommil Just a heads up; I noticed that on #1351 you mentioned rxjava/rx being nasty. I've started writing a wrapper for it, see (link)[https://github.com/yilinwei/rx-cats]  (ignore the name - the cats dep isn't core to the wrapper) which essentially would scalalify it into something more usable (proper variance, using the normal scala functions, type inference goodness).

**Comment by fommil** [comment]
if I fall for fs2 with the drone-dynamic-agent project, I may well be tempted to use it in ensime. I think we can assume cats will be around for the long term and available for new versions of scala (and dotty), so I'm coming around to the idea of depending on a few core typelevel libs... surely swapping akka for fs2 would be ok, right?

**Comment by yilinwei** [comment]
Absolutely; in my opinion the only real reason to go for rxjava is that it tends to have relatively good support across java projects. In any case, swapping between `monix`, `rx` and `fs2` ought to be pretty simple unless you use some more advanced features rather than the more high level combinators like `map` and `flatMap`, since they're all essentially monadic.
